files/migrations.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python3
+"""
+Runs migration scripts for this project
+
+Intended to be triggered by the %post rpm scriptlet rather than manually.
+"""
+
+import logging as log
+import sys
+from importlib.machinery import SourceFileLoader
+from pathlib import Path
+
+import migration_steps
+
+
+class Version:
+    """
+    A sortable software version
+
+    Reads version strings, and sorts by major.minor.patch[.etc[.etc …]]
+    """
+
+    def __init__(self, version):
+        """
+        Read version numbers into lists of ints, discard anything that comes
+        after the first character that is neither a period or a digit
+        """
+        chunks = []
+        # Opting for minimal code over execution speed
+        for part in version.split("."):
+            digits = ""
+            for char in part.split():
+                if char.isdigit():
+                    digits += char
+                else:
+                    # Treat RCs, betas etc. as if it were the full release
+                    break
+            chunks.append(int(digits))
+        self.version = tuple(chunks)
+
+    def __str__(self):
+        return ".".join(str(v) for v in self.version)
+
+
+class Migration(Version):
+    """
+    A sortable runnable migration
+
+    Subclass of Version so that we can sort migrations just as easily as version strings
+    """
+
+    def __init__(self, path):
+        super().__init__(path.name.rsplit(".", 1)[0])
+        self.path = path
+
+    def run_and_update_version_file(self, version_file):
+        """
+        Run this migration
+        """
+
+        migration = SourceFileLoader("migration", str(self.path)).load_module()
```

Comment:
(A small thing: looks like as of python 3.4 `load_module()` is [deprecated](https://docs.python.org/3/library/importlib.html#importlib.abc.Loader.load_module) in favour of `exec_module()`( and `create_module`)

Reply:
I did see this too and was confused by the amount of boilerplate that necessary in the new way. I thought I must have it wrong and just left it at the time, but no, looks like the whole `spec_from_loader` and `module_from_spec` dance is necessary. Updated here too, but also addressed similar stuff in #24. Thanks for flagging!